src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -316,7 +316,7 @@ def flush():
     save(get_config_dir(), _session.generate_payload())
 
     # reset session fields, retaining correlation id and application
-    _session.__init__(correlation_id=_session.correlation_id, application=_session.application)
+    _session.__init__(correlation_id=_session.correlation_id, application=_session.application)  # pylint: disable=unnecessary-dunder-call
```

Comment:
It uses `__init__` to reset fields, so weird.

Reply:
@evelyn-ys for awareness.

Reply:
I don't know why the original author write in this way. Maybe we can try replace with
```
_session = TelemetrySession(correlation_id=_session.correlation_id, application=_session.application)
```